nav2_planner/src/planner_server.cpp

```diff
@@ -522,6 +529,56 @@ PlannerServer::publishPlan(const nav_msgs::msg::Path & path)
   }
 }
 
+void PlannerServer::isPathValid(
+  const std::shared_ptr<nav2_msgs::srv::IsPathValid::Request> request,
+  std::shared_ptr<nav2_msgs::srv::IsPathValid::Response> response)
+{
+  unsigned int mx = 0;
+  unsigned int my = 0;
+
+  response->is_valid = true;
+
+  if (request->path.poses.empty() ) {
+    response->is_valid = false;
+    return;
+  }
+
+  geometry_msgs::msg::PoseStamped current_pose;
+  unsigned int closest_point_index = 0;
+  if (costmap_ros_->getRobotPose(current_pose)) {
+    float current_distance = std::numeric_limits<float>::max();
+    float closest_distance = current_distance;
+    geometry_msgs::msg::Point current_point = current_pose.pose.position;
+    for (unsigned int i = 0; i < request->path.poses.size(); ++i) {
+      geometry_msgs::msg::Point path_point = request->path.poses[i].pose.position;
+
+      current_distance = nav2_util::geometry_utils::euclidean_distance(
+        current_point,
+        path_point);
+
+      if (current_distance < closest_distance) {
+        closest_point_index = i;
+        closest_distance = current_distance;
+      }
+    }
+
+    /**
+     * The lethal check starts at the closest point to avoid points that have already been passed
+     * and may have become occupied
+     */
+    for (unsigned int i = closest_point_index; i < request->path.poses.size(); ++i) {
+      costmap_->worldToMap(
+        request->path.poses[i].pose.position.x,
+        request->path.poses[i].pose.position.y, mx, my);
+      if (static_cast<unsigned int>(costmap_->getCost(mx, my)) >=
+        nav2_costmap_2d::LETHAL_OBSTACLE)
```

Comment:
Hi, 
Shouldn't we check `== nav2_costmap_2d::LETHAL_OBSTACLE` instead of `>=` ?
Because NO_INFORMATION (255) > LETHAL_OBSTACLE (254), I think IsPathValid will not be usable with planners that plans in unknown regions (e.g. to do environment exploration)

Just a suggestion: when experimenting with this PR, I ended to check against `cost == LETHAL_OBSTACLE || cost == INSCRIBED_INFLATED_OBSTACLE` because:
-  If the path point is in the inscribed region then the robot will already collide with the obstacle if following the path, so the path is actually "obstructed"
- I especially wanted to retrigger planning when people crosses the path. But the legs seen by my lidars are represented only by one or two cells on my 0.1m resolution costmap, so the legs had to be exactly on the path to trigger replanning. Using the INSCRIBED_INFLATED_OBSTACLE made it more sensitive and efficient for this application

Reply:
I think it should be `== LETHAL || (!unknown_allowed && UNKNOWN)` type logic. That way if it becomes unknown and we don't allow for unknown travel that case is covered.

Where LETHAL is INSCRIBED_INFLATED_OBSTACLE and LETHAL_OBSTACLE, both. 

So something like 

```
if (!allow_unknown && cost == UKNWOWN) {
  return false
} else if (cost == LETHAL || cost == INSCRIBED) {
  return false
}

```

Reply:
Added in this logic, not sure how to grab the allow_unknow_ parameter. I will need access to the active planner plugin.

Reply:
Done